index.js

```diff
@@ -8,7 +8,7 @@ var crypto = require('crypto');
  * Sign the given `val` with `secret`.
  *
  * @param {String} val
- * @param {String} secret
+ * @param {crypto.CipherKey} secret
```

Comment:
see background in the PR comments, but maybe `@param {String|BinaryLike|KeyObject} secret` would be most clear here?

[technically "String" is already part of "BinaryLike" in the typescript defs that we're borrowing from, but I'd rather err on the side of people knowing a string is still fine here]